neo4j/check.py

```diff
@@ -0,0 +1,148 @@
+        if int(version[0]) > 2:
+        	checkURL = host + ":" + str(port) + "/db/data/transaction/commit"
+        else:
+	        checkURL = host + ":" + str(port) + "/v1/service/metrics"
+        print checkURL
+        tags = instance.get('tags', [])
+        service_check_tags = tags + ['url:%s' % url]
+      
+
+# Neo specific
+# Create payload using built-in Neo4j queryJmx stored procedure
+        try:
+            payload = {"statements" : [{"statement" : "CALL dbms.queryJmx('org.neo4j:*') yield attributes with  keys(attributes) as k, attributes unwind k as row return row, attributes[row]['value'];"}]}
+            headers_sent = {'Content-Type':'application/json','Authorization':'Basic ' + b64Val + '','Content-Type':'application/json'}
+            r = requests.post(checkURL, data=json.dumps(payload),headers=headers_sent)
+
+        except (socket.timeout, socket.error, HttpLib2Error) as e:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
+                               message="Unable to fetch Neo4j stats: %s" % str(e),
+                               tags=service_check_tags)
+            raise
+
+
+        if r.status_code != 200:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
+                               tags=service_check_tags,
+                               message="Unexpected status of %s when fetching Neo4j stats, "
+                               "response: %s" % (r.status_code, r))
+        stats = r.json()
+        self.service_check(
+            self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=service_check_tags)
+
+        for doc in stats['results'][0]['data']:
+            if doc['row'][0].lower() in self.keys:
+                self.gauge(server_name + "." + doc['row'][0], doc['row'][1], tags=tags)
+                
+
+    def timeout_event(self, url, timeout, aggregation_key):
+        self.event({
+            'timestamp': int(time.time()),
+            'event_type': 'http_check',
+            'msg_title': 'URL timeout',
+            'msg_text': '%s timed out after %s seconds.' % (url, timeout),
+            'aggregation_key': aggregation_key
+        })
+
+    def status_code_event(self, url, r, aggregation_key):
+        self.event({
+            'timestamp': int(time.time()),
+            'event_type': 'http_check',
+            'msg_title': 'Invalid reponse code for %s' % url,
+            'msg_text': '%s returned a status of %s' % (url, r.status_code),
+            'aggregation_key': aggregation_key
+        })
+
+    def _get_config(self, instance):
+        self.host = instance.get('neo4j_url', '')
+        self.port = int(instance.get('port', 7474))
+        user = instance.get('user', '')
+        password = str(instance.get('password', ''))
+        connect_timeout = instance.get('connect_timeout', None)
+        server_name = instance.get('server_name','')
+        version = instance.get('neo4j_version', '3.1')
```

Comment:
Added in ability to get version programmatically.
Removed from conf.yaml example.